lib/util.js

```diff
@@ -457,3 +461,157 @@ module.exports = exports = {
+    var config = {
+      events: [],
+      signal: 'SIGUSR2',
+      filename: '',
+      path: '',
+      verbose: false
+    };
+    const report = {
+      setDiagnosticReportOptions(options) {
+        emitExperimentalWarning('report');
+        // reuse the null and undefined checks. Save
+        // space when dealing with large number of arguments.
+        const list = parseOptions(options);
+
+        // flush the stale entries from report, as
+        // we are refreshing it, items that the users did not
+        // touch may be hanging around stale otherwise.
+        config = {};
+
+        // the parseOption method returns an array that include
+        // the indices at which valid params are present.
+        list.forEach((i) => {
+          switch (i) {
+            case REPORTEVENTS:
+              if (Array.isArray(options.events))
+                config.events = options.events;
+              else
+                throw new ERR_INVALID_ARG_TYPE('events',
+                                               'Array',
+                                               options.events);
+              break;
+            case REPORTSIGNAL:
+              if (typeof options.signal !== 'string')
+                throw new ERR_INVALID_ARG_TYPE('signal',
+                                               'String',
+                                               options.signal);
+              process.removeListener(config.signal, handleSignal);
+              if (config.events.includes('signal'))
+                process.on(options.signal, handleSignal);
+              config.signal = options.signal;
+              break;
+            case REPORTFILENAME:
+              if (typeof options.filename !== 'string')
+                throw new ERR_INVALID_ARG_TYPE('filename',
+                                               'String',
+                                               options.filename);
+              config.filename = options.filename;
+              break;
+            case REPORTPATH:
+              if (typeof options.path !== 'string')
+                throw new ERR_INVALID_ARG_TYPE('path', 'String', options.path);
+              config.path = options.path;
+              break;
+            case REPORTVERBOSE:
+              if (typeof options.verbose !== 'string' &&
+                  typeof options.verbose !== 'boolean')
+                throw new ERR_INVALID_ARG_TYPE('verbose',
+                                               'Booelan | String' +
+                                               ' (true|false|yes|no)',
+                                               options.verbose);
+              config.verbose = options.verbose;
+              break;
+          }
+        });
+        // upload this new config to C++ land
+        nr.syncConfig(config, true);
+      },
+
+
+      triggerReport(file, err) {
+        emitExperimentalWarning('report');
+        if (err == null) {
+          if (file == null)
```

Comment:
Style nit: Can you add `{ … }` when the conditional’s body spans more than one line?